src/Configuration.ts

```diff
@@ -0,0 +1,491 @@
+/*
+ * Copyright 2020-2023 Delft University of Technology and SynTest contributors
+ *
+ * This file is part of SynTest Framework - SynTest Core.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Yargs = require("yargs");
+import * as path from "path";
+import shell = require("shelljs");
+
+export let CONFIG: ArgumentsObject;
+export class Configuration {
+  initializeConfigSingleton<T extends ArgumentsObject>(argumentValues: T) {
+    if (CONFIG) {
+      throw Error("Already initialized the config singleton!");
+    }
+
+    CONFIG = argumentValues;
+  }
+
+  private programName: string;
```

Comment:
Does it make sense to have the program name in the configuration? This is not configurable

Reply:
I only need it there to create the --help output. I removed it from the constructor and added it to the specific function that requires it.

Reply:
Sounds good!